src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,46 @@
+package duke;
+
+import duke.command.Command;
+
+/**
+ * A chatbot that helps you keep track of tasks.
+ */
+public class Duke {
+    private final Storage storage;
+    private final Ui ui;
+    private TaskList tasks;
+
+    /**
+     * Constructor for a duke.Duke instance.
+     *
+     * @param filePath The location of the save file to be created at and read from.
+     */
+    public Duke(String filePath) {
```

Comment:
I like that the constructor and the getResponse method are appropriately sized and do not exhibit deep nesting!